### PR TITLE
ICRC Identifier in ICP load accounts

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

Extend `icp-accounts.service` load accounts function to map ICRC account identifiers.

# Changes

- new type `ICPAccount` that extends `Account` to populate ICP and Icrc identifiers for ICP/NNS
- map ICRC or ICP identifier in `loadAccounts` function dependending if flag is on or not

### CODE REVIEW

File: frontend/src/lib/utils/accounts.utils.ts [hunk 1]
```diff
@@ -229,3 +233,37 @@ export const sumAccounts = (
 
 export const hasAccounts = (accounts: Account[]): boolean =>
   accounts.length > 0;
+
+export const toIcpAccountIdentifier = (
```
Comment (lmuntaner) [inline frontend/src/lib/utils/accounts.utils.ts:237]: Is this used anywhere?

File: frontend/src/lib/services/icp-accounts.services.ts [hunk 2]
```diff
@@ -98,8 +107,19 @@ export const loadAccounts = async ({
     (type: AccountType) =>
     async (
       account: AccountDetails | HardwareWalletAccountDetails | SubAccountDetails
-    ): Promise<Account> => ({
-      identifier: account.account_identifier,
+    ): Promise<IcpAccount> => ({
+      identifier: icrcEnabled
```
Comment (lmuntaner) [inline frontend/src/lib/services/icp-accounts.services.ts:111]: Why do you think we should pass the feature flag as a parameter?

I'm not sure I understand the value of all these changes.

File: frontend/src/lib/utils/accounts.utils.ts (hunk 1 above)
Comment (peterpeterparker) [reply frontend/src/lib/utils/accounts.utils.ts:237]: Yes, `icp-accounts.services`

File: frontend/src/lib/services/icp-accounts.services.ts (hunk 2 above)
Comment (peterpeterparker) [reply frontend/src/lib/services/icp-accounts.services.ts:111]: If we want to dynamically switch / activate / deactivate the feature flag, we will need to reactively reload the accounts, therefore we would need to subscribe to it in components.

But, not strong opinion. If you rather like to read the flag imperatively in `services` (`get(store)`) for sure I can adapt. I'm fine with both.

Comment (peterpeterparker) [reply frontend/src/lib/services/icp-accounts.services.ts:112]: Discussed per PM, we go for the imperative access to the store in the services.

@lmuntaner done.

File: frontend/src/lib/utils/accounts.utils.ts (hunk 1 above)
Comment (peterpeterparker) [reply frontend/src/lib/utils/accounts.utils.ts:237]: Ah no you are right, not in this PR. Let me provide it in another PR.

**Review by lmuntaner — APPROVED** [review]
Thanks for the changes! LGTM!

**Pull request merged**